### PR TITLE
(feat) require payments before picking lab orders

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary
This PR enhances lab result retrieval by ensuring it occurs only when the corresponding test is paid for, in emergency cases, or during inpatient visits.

### CODE REVIEW

File: src/queue-list/pick-lab-request-menu.component.tsx [hunk 1]
```diff
@@ -22,13 +27,28 @@ const PickLabRequestActionMenu: React.FC<PickLabRequestActionMenuProps> = ({
     });
   }, [order]);
 
+  const overflowMenuItemLabel = shouldPayBill
+    ? t("payBill", "Pay Bill")
```
Comment (ojwanganto) [inline src/queue-list/pick-lab-request-menu.component.tsx:31]: This is an intelligent approach. Thanks. Maybe we change the text to read "Pending bill" or something.

File: src/bills/bill.resource.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  return { shouldPayBill: hasPendingBill, isLoading: false };
+};
+
+const usePatientQueue = (patientUuid: string) => {
+  const url = `/ws/rest/v1/visit-queue-entry?patient=${patientUuid}`;
+  const { data, isLoading, error } = useSWR<{
+    data: { results: Array<QueueEntry> };
+  }>(url, openmrsFetch);
+  const isEmergencyPatient =
+    data?.data?.results?.[0]?.queueEntry?.priority?.display === "Emergency";
```
Comment (ojwanganto) [inline src/bills/bill.resource.tsx:57]: Will it make any difference moving the "Emergency" into a variable? Am not sure

**Review by ojwanganto — APPROVED** [review]
Thanks, Don

File: src/queue-list/pick-lab-request-menu.component.tsx (hunk 1 above)
Comment (donaldkibet) [reply src/queue-list/pick-lab-request-menu.component.tsx]: Sure made the changes

File: src/bills/bill.resource.tsx (hunk 2 above)
Comment (donaldkibet) [reply src/bills/bill.resource.tsx]: For consistency should be done

**Pull request merged**